### PR TITLE
feat: hand-labelled FLOWS_TO ground-truth corpus with per-metric eval scope reporting

### PR DESCRIPTION
**Pull request by vitali87**
First slice of #1190 (does not close it): the FLOWS_TO ground-truth stage 1 plus the honest per-metric scope reporting. Nothing graded FLOWS_TO anywhere before this; it was the newest and subtlest edge type with zero ground truth.

## What's in

- **`evals/flows_corpus/`**: a hand-labelled per-language corpus (Python deep walk; JavaScript, Go, Java, Scala lean walk; Dart leaf path). Each language directory indexes as its own repo and contains one ENV -> STDOUT flow plus one deliberate non-flow; `expected.json` holds the labels as `(source_resource, sink_resource, kind)` triples.
- **`evals/flow_ground_truth.py`**: scores emitted FLOWS_TO triples against the labels per language (precision/recall, `evals/results/flows.csv`, rich table via `python -m evals.flow_ground_truth`). It indexes a temp COPY of each fixture dir so GraphUpdater cache artifacts never dirty the corpus.
- **`test_eval_flows.py`**: pins precision == recall == 1.0 per language, and that the label set covers exactly the six corpus languages. A recall regression means a flow-walk break; a precision regression means a fabricated flow the labeller did not sanction.
- **Honest scope reporting**: a "Per-metric language scope" table in `evals/README.md` stating exactly which languages each metric family measured (the public 1.0/1.0 imports/inheritance numbers are Python-only, call retrieval is file-level simple-name, etc.), so no 1.0 reads wider than it was measured.

## The corpus paid for itself on its first run

1. **Scala emitted zero FLOWS_TO edges** despite `flow_covered: true` (from #1322): the descriptor's `subscript_type` was wired to `call_expression`, so the flow walk's member/subscript branch swallowed every nested source call before the call branch could taint it. Fixed here (`subscript_type=type_identifier`, a node that never occurs in expression position, since Scala spells subscripts as apply calls) and pinned by the new `test_flow_scala_edges.py` (direct nesting, val-bound flow, and a constant non-flow).
2. **The Python deep walk misses the `os.environ["X"]` subscript source** (`os.getenv`/`os.environ.get` flow correctly). Filed separately as a follow-up issue rather than papered over; the corpus fixture uses `os.getenv`.

## Out of scope (stays on #1190)

Dart/Ruby structural oracles, resolved-edge grading beyond Python, qn-level call grading, and compiler-generated flow ground truth at scale (Roslyn AnalyzeDataFlow, go/ssa) — those need toolchains and are staged next.

## Tests

`test_eval_flows.py` + `test_flow_scala_edges.py` green; flow battery (flow verdict, Dart/PHP flow edges, Scala io) 57 passed; harness run: 1.0/1.0 across all six languages.

## Summary by CodeRabbit

- **New Features**
  - Added a hand-labelled flow-analysis evaluation corpus covering Python, JavaScript, Go, Java, Scala, and Dart.
  - Added a command-line evaluation tool with precision and recall reporting and CSV export.
  - Documented supported language coverage and expected environment-variable-to-output flow mappings.

- **Bug Fixes**
  - Improved Scala analysis for nested and locally assigned environment-variable flows while avoiding false positives from constants and method calls.

- **Tests**
  - Added cross-language regression checks and expanded Scala edge-case coverage, achieving complete precision and recall across the evaluation corpus.

### CODE REVIEW

File: evals/flows_corpus/python/app.py [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+
+def leak():
+    secret = os.getenv("TOKEN")
+    print(secret)
+
+
+def safe():
+    fixed = "constant"
+    return fixed
```
[the remark this answers is not in the document]
Comment (vitali87) [reply evals/flows_corpus/python/app.py]: Applied in ac9784ca to ALL six languages, not just Python: every fixture's safe() now sends its constant into the language's stdout sink (print/console.log/fmt.Println/System.out.println/println) with no expected triple, so a fabricated flow for clean data now costs precision in every language. Verified 1.0/1.0 across the corpus after the change.

File: evals/flows_corpus/python/app.py [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import os
+
+
+def leak():
+    secret = os.getenv("TOKEN")
+    print(secret)
```
[the remark this answers is not in the document]
Comment (vitali87) [reply evals/flows_corpus/python/app.py]: This is the ground-truth fixture whose entire purpose is to BE that leak: the corpus asserts the flow analysis detects exactly this clear-text ENV-to-stdout flow (evals/flows_corpus, #1190 stage 1). It is never imported by product code. Dismissed the code-scanning alert as used-in-tests with that rationale.

[the remark this answers is not in the document]
Comment (vitali87) [reply evals/flows_corpus/python/app.py]: Dismissed as used-in-tests (same rationale as alert 81: the fixture's entire purpose is to BE this leak so the flow analysis can be graded on detecting it), and 8deb0802 renames the fixture variables to neutral names (value) in the CodeQL-analyzed languages so corpus edits stop re-keying fresh alerts; the ground truth keys on the source call, never the variable name.
